app/models/tenant.rb

```diff
@@ -1,3 +1,29 @@
 class Tenant < ApplicationRecord
   validates :external_tenant, :uniqueness => true, :presence => true
+
+  after_initialize :setup_settings, :unless => proc { settings.nil? }
+
+  def add_setting(name, value)
+    settings[name] = value
+    save!
+  end
+
+  def update_setting(name, value)
+    add_setting(name, value)
```

Comment:
Does this have the potential to just add a new setting every time if the `name` doesn't exist?

Reply:
Yep you're absolutely right, I'll add validation to make sure the key is there before updating.